File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        n=len(prices)
        minBuy=float("inf")
        maxProfit=0

        for i in range(n):
            if prices[i]<minBuy:
                minBuy=prices[i]
            else:
                maxProfit=max(maxProfit,prices[i]-minBuy)
        
        return maxProfit

        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n=prices.length;
        int minBuy=Integer.MAX_VALUE;
        int maxProfit=0;

        for(int i=0;i<n;i++){
            if(prices[i]<minBuy){
                minBuy=prices[i];
            }
            else{
                maxProfit=Math.max(maxProfit,prices[i]-minBuy);
            }

        }
        return maxProfit;
        
    }
}